### PR TITLE
🏅 OOP school library: decorate a class 🏅

### PR DESCRIPTION
**Pull request by kambereBr**
## 📝 General requirements

- ✅ No linters errors
- ✅ Used the correct GitFlow.
#### Interface
- ✅ Created a class `Nameable`.
   - Implemented a method called `correct_name` that will raise a `NotImplementedError`.

#### Turned Person class to Nameable
- ✅ Made sure that your Person class inherits from Nameable
- ✅ Made sure that this class has a method `correct_name` implemented. It should simply return the name attribute.

#### Prepared base Decorator
- ✅ Made sure that it inherits from Nameable.
- ✅ In the constructor assigned a nameable object from params to an instance variable.
- ✅ Implemented the `correct_name` method that returns the result of the correct_name method of the `@nameable`.

#### Prepared CapitalizeDecorator and TrimmerDecorator
- ✅ For the CapitalizeDecorator:
    - Created a class that inherits from the base Decorator class.
    - Implemented a method `correct_name` that capitalizes the output of `@nameable.correct_name`.
- ✅ For the TrimmerDecorator:
    - Created a class that inherits from the base Decorator class.
    - Implemented a method `correct_name` that makes sure that the output of `@nameable.correct_name` has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

File: app/models/capitalize_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require_relative 'decorator'
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name
+  end
```
Comment (shahadat3669) [inline app/models/capitalize_decorator.rb]: - [ ] 👍 Great job on implementing the `correct_name` method! However, as per the project requirement, this method should return the name with capitalization. Currently, it's returning the name without capitalization. For example, if I use `CapitalizeDecorator.new(person)` where the person's name is `maximilianus`, it should return `Maximilianus`. Currently, it's returning `maximilianus`. Please update your code to ensure that the `correct_name` method capitalizes the name as required. Keep up the good work! 📝👏

| Your | Expected|
| --- | --- |
|  image| image|

**Review by shahadat3669 — CHANGES_REQUESTED** [review]
Hi @kambereBr, 

Great job so far! 👏 You've done well, and you're almost there for the next project ⏭️, but there are a few issues 🐛 that you still need to work on.




## Don't Worry 🙏

It's okay to have bugs in your code. 🆗 Please see this code review as an opportunity to learn practical coding skills and discover new tips that can improve your code quality 💯 and help you **get even better** 💼.

**I will guide you on how to address these areas down below 👇.**

## Highlights
✅ Has a `Namable` class with a method 'correct_name' rasing an error.
✅ `Person` class inherits from `Nameable` class.
✅ `TrimmerDecorator` class working as expected 👍 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadata3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

## ⚠️ WARNING ⚠️

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/capitalize_decorator.rb (hunk 1 above)
Comment (kambereBr) [reply app/models/capitalize_decorator.rb]: Thanks dear @shahadat3669 for your review 👏

**Review by Munsa1 — APPROVED** [review]
Hi @kambereBr, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Made Requested Changes By The Previous Reviewer :heavy_check_mark: 
Correct Output :+1: 
 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by kambereBr** [comment]
Thanks dear @Munsa1 👏

**Pull request merged**